framework/src/st_things/things_stack/logging/things_logger.c

```diff
@@ -130,3 +133,252 @@ void things_logv(things_log_level_e level, const char *tag, const char *func_nam
 	va_end(args);
 	things_log(level, tag, func_name, line_num, buffer);
 }
+
+struct num_str {
+	int number;
+	char *str;
+};
+
+/**
+ * CoAP Option Numbers
+ * http://www.iana.org/assignments/core-parameters/core-parameters.xhtml
+ */
+static struct num_str _coap_options[] = {
+	{ 0, "Reserved" },
+	{ 1, "If-Match" },
+	{ 3, "Uri-Host" },
+	{ 4, "ETag" },
+	{ 5, "If-None-Match" },
+	{ 6, "Observe" },
+	{ 7, "Uri-Port" },
+	{ 8, "Location-Path" },
+	{ 11, "Uri-Path" },
+	{ 12, "Content-Format" },
+	{ 14, "Max-Age" },
+	{ 15, "Uri-Query" },
+	{ 17, "Accept" },
+	{ 20, "Location-Query" },
+	{ 23, "Block2" },
+	{ 27, "Block1" },
+	{ 28, "Size2" },
+	{ 35, "Proxy-Uri" },
+	{ 39, "Proxy-Scheme" },
+	{ 60, "Size1" },
+	{ 128, "Reserved" },
+	{ 132, "Reserved" },
+	{ 136, "Reserved" },
+	{ 140, "Reserved" },
+	{ 258, "No-Response" },
+	{ 2049, "OCF-Accept-Content-Format-Version" },
+	{ 2053, "OCF-Content-Format-Version" },
+};
+
+static const char *_coap_opt_name(int number)
+{
+	unsigned int i;
+
+	for (i = 0; i < sizeof(_coap_options) / sizeof(struct num_str); i++) {
+		if (_coap_options[i].number == number)
+			return _coap_options[i].str;
+	}
+
+	return "Unknown";
+}
+
+static void _ll(things_log_level_e level, const char *tag, const char *func_name, const int16_t line_num,
+							OCStringLL *ll)
+{
+	OCStringLL *cur = ll;
+
+	while (cur) {
+		things_logv(level, tag, func_name, line_num, "\t\t  %s ", cur->value);
+		cur = cur->next;
+	}
+}
+
+static void _ll_rep_value(things_log_level_e level, const char *tag, const char *func_name, const int16_t line_num,
+							OCRepPayloadValue *values)
+{
+	OCRepPayloadValue *cur = values;
+
+	while (cur) {
+		switch (cur->type) {
+		case OCREP_PROP_INT:
+			things_logv(level, tag, func_name, line_num, "\t  {\"%s\":  %zd} ", cur->name, cur->i);
+			break;
+		case OCREP_PROP_DOUBLE:
+			things_logv(level, tag, func_name, line_num, "\t  {\"%s\":  %f} ", cur->name, cur->d);
+			break;
+		case OCREP_PROP_BOOL:
+			things_logv(level, tag, func_name, line_num, "\t  {\"%s\":  %d} ", cur->name, cur->b);
+			break;
+		case OCREP_PROP_STRING:
+			things_logv(level, tag, func_name, line_num, "\t  {\"%s\":  \"%s\"} ", cur->name, cur->str);
+			break;
+		case OCREP_PROP_BYTE_STRING:
+			things_logv(level, tag, func_name, line_num, "\t  {\"%s\":  len=%zd [...]} ", cur->name, cur->ocByteStr.len);
+			break;
+		case OCREP_PROP_OBJECT:
+			things_logv(level, tag, func_name, line_num, "\t  {\"%s\":  {0x%p}}", cur->name, cur->obj);
+			break;
+		case OCREP_PROP_ARRAY:
+			things_logv(level, tag, func_name, line_num, "\t  {\"%s\":  [type=%d]}", cur->name, cur->arr.type);
+			break;
+		default:
+			things_logv(level, tag, func_name, line_num, "\t  {\"%s\":  unknown} ");
+			break;
+		}
+
+		cur = cur->next;
+	}
+}
+
+static void things_log_payload_flag(things_log_level_e level, const char *tag, const char *func_name, const int16_t line_num,
```

Comment:
What is the difference between a function with "_" under bar and a function without function name?

_coap_opt_name ()
_ll ()
_ll_rep_value ()

Would not it be a good idea to prefix the under bar with static functions like this?

Reply:
Well.. My thingking those function will use only in this file spaces. So, I copied function name from function name of slsi debug function's name. 
However, It looks not good.